jar-connector/src/main/java/com/sixsq/slipstream/connector/ConnectorBase.java

```diff
@@ -294,18 +294,31 @@ protected String getPrivateSshKeyFileName() throws ConfigurationException, Valid
         return privateSshKeyFile;
     }
 
+    public static String getServerPublicSshKeyFilename() throws ConfigurationException, ValidationException {
+		return Configuration.getInstance().getProperty(ServiceConfiguration.CLOUD_CONNECTOR_ORCHESTRATOR_PUBLICSSHKEY);
+    }
+
     private String getUserPublicSshKey(User user) throws ValidationException, IOException {
 		return user.getParameter(
 				ExecutionControlUserParametersFactory.CATEGORY + "." + UserParametersFactoryBase.SSHKEY_PARAMETER_NAME)
 				.getValue();
     }
 
+    private String getUserOrServerPublicSshKey(User user) throws ValidationException, IOException {
```

Comment:
Generally, is there a reason to insert the server's SSH key rather than just using a dummy value?  The problem with inserting a working key is that it looks like we're providing a backdoor into the machine.